ansible_collections/arista/avd/docs/input-variable-validation-BETA.md

```diff
@@ -0,0 +1,243 @@
+# Validation of Input Variables (BETA)
+
+!!! BETA
+    - The Type Conversion and Validation task is currently only enabled by default on `eos_cli_config_gen`. Run with `--tags validate` to test validation on `eos_designs`.
+    - Schemas for all input data models are still under development
+    - Some schema validations are not implemented yet.
+
+`eos_designs` and `eos_cli_config_gen` supports a wide range of input variables described under the role documentation sections.
+
+Internally the supported data models are described in a proprietary "AVD Schema" format, which is used to perform validation of
+the input variables at run-time. The same schema is also used to generate the role documentation describing the supported data
+models.
+
+For details on the schema format see the [Schema Details](#schema-details) section below.
+
+## Automatic Type Conversion
+
+The validation of input variables is performed by the `arista.avd.validate` Action Plugin. The plugin also supports automatic
+conversion of variable types, if configured in the schema. The supported types that can be converted from depends on the type to
+be converted to. The current implementation supports the following automatic conversions:
+
+| From (`convert_types`) | To (`type`) |
+| ---------------------- | ----------- |
+| `bool`, `str` | `int` |
+| `int`, `str` | `bool` |
+| `bool`, `int` | `str` |
+| `dict`* | `list` |
+
+*) Conversion from `dict` to `list` is only supported when `primary_key` is set on the `list` schema.
+
+An example of input variable conversion is `bgp_as`. `bgp_as` is expected as a string (`str`) since a 32-bit AS numbers can be
+written with dot-notation ex. `"65001.10000"`. Most deployments only use 16-bit AS numbers ex. `bgp_as: 65001`, and this value
+will be interpreted as an integer in YAML, unless it is enclosed in quotes ex. `bgp_as: "65001"`. In the schema for `bgp_as` the
```

Comment:
Consider an alternate to ex in technical document. See https://developers.google.com/style/format-examples